File: python-algorithm/leetcode/problem_89.py
"""89. Gray Code
https://leetcode.com/problems/gray-code/
"""
from typing import List


def gray_code(n: int) -> List[int]:
    ans = []
    size = 1 << n
    for i in range(size):
        ans.append(i ^ (i >> 1))
    return ans

File: go-algorithm/leetcode/problem89.go
package main

// 89. Gray Code
// https://leetcode.com/problems/gray-code/

func grayCode(n int) []int {
	ans := make([]int, 0)
	size := 1 << n
	for i := 0; i < size; i++ {
		ans = append(ans, i^(i>>1))
	}
	return ans
}
